src/treeview_provider/modelTreeviewProvider.ts

```diff
@@ -500,3 +515,16 @@ export class DocumentationTreeview extends DocumentationTreeviewProvider {
 
 @provideSingleton(IconActionsTreeview)
 export class IconActionsTreeview extends IconActionsTreeviewProvider {}
+
+// Find appropriate a model from file content (if YAML) or from a file name (otherwise)
+export function lookupModelByEditorContent(
+  nodeMetaMap: NodeMetaMap,
+  document: TextDocument,
+): NodeMetaData | undefined {
+  const modelCandidateName =
+    document.languageId === "yaml" &&
+    getCurrentlySelectedModelNameInYamlConfig()
+      ? getCurrentlySelectedModelNameInYamlConfig()
+      : path.parse(document.fileName).name;
```

Comment:
This is not necessarily correct as the name might not be a valid model name or even just wrong while  parsing yaml should be correct, but I guess it is OK to have as fallback.